internal/host/stopinstallation.go

```diff
@@ -10,21 +14,33 @@ import (
 
 type stopInstallationCmd struct {
 	baseCmd
+	instructionConfig InstructionConfig
 }
 
-func NewStopInstallationCmd(log logrus.FieldLogger) *stopInstallationCmd {
+func NewStopInstallationCmd(log logrus.FieldLogger, instructionConfig InstructionConfig) *stopInstallationCmd {
 	return &stopInstallationCmd{
-		baseCmd: baseCmd{log: log},
+		baseCmd:           baseCmd{log: log},
+		instructionConfig: instructionConfig,
 	}
 }
 
 func (h *stopInstallationCmd) GetStep(ctx context.Context, host *models.Host) (*models.Step, error) {
-	step := &models.Step{
-		StepType: models.StepTypeExecute,
-		Command:  "/usr/bin/podman",
-		Args: []string{
-			"kill", "--all",
-		},
+	step := &models.Step{}
+	step.StepType = models.StepTypeExecute
+	step.Command = "/usr/bin/podman"
+	cmdArgs := []string{"kill", "--all"}
+
+	// added to run upload logs if we are in error or cancelled state. Stop all and gather logs
+	// will return same exit code as stop command command
+	if host.LogsCollectedAt == strfmt.DateTime(time.Time{}) {
+		logsCommand, err := CreateUploadLogsCmd(host, h.instructionConfig.ServiceBaseURL,
+			h.instructionConfig.InventoryImage, h.instructionConfig.SkipCertVerification, true)
+		if err != nil {
+			h.log.WithError(err).Error("Failed to create logs upload command")
+		}
+		step.Command = "bash"
+		cmdArgs = []string{"-c", "podman " + strings.Join(cmdArgs, " ") + " || " + logsCommand}
```

Comment:
the logs command won't run if the podman command is successful

Reply:
right, my fault. :) it always fails and i missed it